pkg/core/native/management.go

```diff
@@ -17,19 +30,288 @@ type Management struct {
 	var m = &Management{ContractMD: *interop.NewContractMD(managementName)}
 
+	desc := newDescriptor("getContract", smartcontract.ArrayType,
+		manifest.NewParameter("hash", smartcontract.Hash160Type))
+	md := newMethodAndPrice(m.getContract, 1000000, smartcontract.ReadStates)
+	m.AddMethod(md, desc)
+
+	desc = newDescriptor("deploy", smartcontract.ArrayType,
+		manifest.NewParameter("script", smartcontract.ByteArrayType),
+		manifest.NewParameter("manifest", smartcontract.ByteArrayType))
+	md = newMethodAndPrice(m.deploy, 0, smartcontract.WriteStates)
+	m.AddMethod(md, desc)
+
+	desc = newDescriptor("update", smartcontract.VoidType,
+		manifest.NewParameter("script", smartcontract.ByteArrayType),
+		manifest.NewParameter("manifest", smartcontract.ByteArrayType))
+	md = newMethodAndPrice(m.update, 0, smartcontract.WriteStates)
+	m.AddMethod(md, desc)
+
+	desc = newDescriptor("destroy", smartcontract.VoidType)
+	md = newMethodAndPrice(m.destroy, 10000000, smartcontract.WriteStates)
+	m.AddMethod(md, desc)
+
 	return m
 }
 
+// getContract is an implementation of public getContract method, it's run under
+// VM protections, so it's OK for it to panic instead of returning errors.
+func (m *Management) getContract(ic *interop.Context, args []stackitem.Item) stackitem.Item {
+	hashBytes, err := args[0].TryBytes()
+	if err != nil {
+		panic(err)
+	}
+	hash, err := util.Uint160DecodeBytesBE(hashBytes)
+	if err != nil {
+		panic(err)
+	}
+	ctr, err := m.GetContract(ic.DAO, hash)
+	if err != nil {
+		panic(err)
+	}
+	return contractToStack(ctr)
+}
+
+// GetContract returns contract with given hash from given DAO.
+func (m *Management) GetContract(d dao.DAO, hash util.Uint160) (*state.Contract, error) {
+	contract := new(state.Contract)
+	key := makeContractKey(hash)
+	err := getSerializableFromDAO(m.ContractID, d, key, contract)
+	if err != nil {
+		return nil, err
+	}
+	return contract, nil
+}
+
+func getLimitedSlice(arg stackitem.Item, max int) ([]byte, error) {
+	_, isNull := arg.(stackitem.Null)
+	if isNull {
+		return nil, nil
+	}
+	b, err := arg.TryBytes()
+	if err != nil {
+		return nil, err
+	}
+	l := len(b)
+	if l == 0 {
+		return nil, errors.New("empty")
+	} else if l > max {
+		return nil, fmt.Errorf("len is %d (max %d)", l, max)
+	}
+
+	return b, nil
+}
+
+// getNefAndManifestFromItems converts input arguments into NEF and manifest
+// adding appropriate deployment GAS price and sanitizing inputs.
+func getNefAndManifestFromItems(args []stackitem.Item, v *vm.VM) (*nef.File, *manifest.Manifest, error) {
+	nefBytes, err := getLimitedSlice(args[0], math.MaxInt32) // Upper limits are checked during NEF deserialization.
+	if err != nil {
+		return nil, nil, fmt.Errorf("invalid NEF file: %w", err)
+	}
+	manifestBytes, err := getLimitedSlice(args[1], manifest.MaxManifestSize)
+	if err != nil {
+		return nil, nil, fmt.Errorf("invalid manifest: %w", err)
+	}
+
+	if !v.AddGas(int64(StoragePrice * (len(nefBytes) + len(manifestBytes)))) {
```

Comment:
MaxScriptLength for NEF file was changed from 1024*1024 to 512*1024.

Reply:
It's ironical that `MaxTransactionSize` is still 100K...

Reply:
Fixed.